docs/boilerplate.md

```diff
@@ -0,0 +1,126 @@
+# Understanding Boilerplate in Oryx
+
+"How to Create Maintainable and Upgradeable Applications"
+
+## Introduction
+
+Boilerplate refers to the _template_ code that is used to generate an application code that can be further customized. Changing boilerplate is convenient at first sight as the code is at hand, generated in your project repository. However, when upgrading to newer versions of the original code, it becomes challenging. If you've customize the boilerplate code and want to reapply a new version of the boilerplate, you have to merge the customizations with the boilerplate during every update to a new version of the application. This is a time-consuming and error-prone process that can slow down your development process and increase the risk of bugs.
+
+## How we prevent boilerplate code
+
+In Oryx, we use a few tactics to prevent boilerplate code. These tactics include:
+
+1. bootstrap the application from npm packages rather than source.
+2. Expose an appBuilder function to setup the [app orchestrator](./app-orchestrator.md) conveniently.
+3. Provide [presets](./presets.md) for the standard application setup, including the feature sets and UI themes.
+4. Provide the source code in a public repository ([github.com/spryker/oryx](https://github.com/spryker/oryx)) to enable developers to customize the code. The code is implemented with TypeScript and the types and interfaces are documented where ever needed to provide a good developer experience.
```

Comment:
Maybe it's worth to describe how exactly it helps in preventing boilerplate code?

Reply:
I'll give it a try, but not sure if it helps.